js/apps/system/_admin/aardvark/APP/react/src/views/services/useFetchServices.ts

```diff
@@ -0,0 +1,27 @@
+import useSWR from "swr";
+import { getAardvarkRouteForCurrentDb } from "../../utils/arangoClient";
+import { ServiceDescription } from "./Service.types";
+
+export const useFetchServices = () => {
+  const { data } = useSWR("/services", async () => {
+    const route = getAardvarkRouteForCurrentDb("foxxes");
+    const data = await route.get();
+    // fetch conf & deps for each service
+    const services = data.body.map(async (service: ServiceDescription) => {
+      const config = await route.get("config", {
+        mount: service.mount
+      });
+      const deps = await route.get("deps", {
+        mount: service.mount
+      });
+      return {
+        ...service,
+        config: config.body,
+        deps: deps.body
+      };
+    });
+    const finalServices = await Promise.all(services);
+    return finalServices as ServiceDescription[];
```

Comment:
Any particular need of assertion?

Reply:
yes, without this the type is `any[]`

Reply:
How about using `Promise.all<Type>(services)`.. Still the same issue?

Reply:
yes, that will work too!